opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -1047,8 +1041,8 @@ namespace Opm
 
                 const double thp = this->calculateThpFromBhp(well_state, well_rates_bhp_limit, bhp_limit, deferred_logger);
                 const double thp_limit = this->getTHPConstraint(summaryState);
-
-                if (thp < thp_limit) {
+                const double temp = thp < thp_limit;
+                if ( (this->isProducer() && temp < 0.) || (this->isInjector() && temp > 0.) ) {
```

Comment:
This too is a very unexpected use of a floating-point variable.  Furthermore, this definition of `temp` shadows the [outer](https://github.com/OPM/opm-simulators/blob/8f2261b4a25f5f0b6475af6d70599e83b723c432/opm/simulators/wells/StandardWell_impl.hpp#L1027) object with the same name which further increases suspicion.